packages/stack/test-runner/src/lib/index.js

```diff
@@ -131,6 +141,82 @@ class TestRunner {
       cb(null, [filePath]);
     });
   }
+
+  async checkDeploymentOptions(config, options, cb = () => {}) {
+    let resetServices = false;
+    const blockchainConfig = config.blockchain || {};
+    let {host, port, type, protocol} = blockchainConfig.endpoint ? deconstructUrl(blockchainConfig.endpoint) : {};
+    const accounts = blockchainConfig.accounts;
+
+    if (host && port && !['rpc', 'ws'].includes(type)) {
+      return cb(__("contracts config error: unknown deployment type %s", type));
+    }
+
+    if (options.coverage && type === 'rpc') {
+      this.logger.warn(__('Coverage does not work with an RPC node'));
```

Comment:
I'm not sure this is true. We fetch blocks after the coverage finishes running and parse the events then. Am I missing something?

Reply:
I didn't write this warning, it's a legacy one, but you might be right that it's no longer needed. I know it was indeed true before because we were aggregating the events as the tests ran, meaning that we needed WS. Can you confirm that we don't use `event.on` or `once` anymore but only `pastEvents`?

Reply:
Coverage actually [fetches blocks all the way from 0](https://github.com/embark-framework/embark/blob/043ccc0a24725c5ed0f586ac8b7acdc402cfe38a/packages/plugins/coverage/src/lib/index.ts#L91-L96), parses events into their respective contract files and generates coverage from there. It doesn't require live events.

Reply:
Ok perfect. I can remove the warning then, though Ws is always better haha